### PR TITLE
add developer documention

### LINKED ISSUE #33: add "HOWTO test the code(unit test and e2e test)"
We need a document for anyone wanting to either write code which directly accesses the Cyclone API, or to contribute directly to the Cyclone project. It should include:

- [x] API doc(Just move the  API doc to this document)
- [x] The process of developing and contributing code to the Cyclone project. (Same as https://github.com/kubernetes/kubernetes/blob/master/docs/devel/README.md)
- [x] HOWTO setup dev environment
- [ ] HOWTO test the code(unit test and e2e test)

### PR DESCRIPTION
**Pull request by gaocegege**
fixes #33 

Signed-off-by: Ce Gao <ce.gao@outlook.com>

### CODE REVIEW

**Comment by gaocegege** [comment]
PTAL

**Comment by superxi911** [comment]
@Wumengna0402 PTAL

File: docs/developer-guide.md [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+
+This section of the documentation contains a guide for users who want to contribute code or documentation to Cyclone.
+
+## The process of developing the Cyclone project
+
+Now the Cyclone project is hosted on Github, so we follow the [github flow](https://guides.github.com/introduction/flow/), you could learn it in 5 minutes:)
+
+## Setting up your dev environment
+
+We have writen some bash scripts to help you start contributing code or documentation to the Cyclone project.
```
Comment (superxi911) [inline docs/developer-guide.md]: writen -> written

File: docs/developer-guide.md [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
+
+If you want to start a cyclone server for dev, you could run:
+
+```shell
+./scripts/local-up.sh
+```
+
+The script would run all the dependencies in docker containers, compile and run cyclone server **locally**. This way is more hackable than run all services in containers.
```
Comment (superxi911) [inline docs/developer-guide.md]: than running all services

File: docs/developer-guide.md [hunk 3]
```diff
@@ -1,5 +1,35 @@
...
+
+Notice that if your docker daemon is running in a docker machine, maybe you need to do some additional work, such as port forwarding and so on.
```
Comment (superxi911) [inline docs/developer-guide.md]: porting forwarding

File: docs/developer-guide.md [hunk 4]
```diff
@@ -12,14 +42,15 @@
 - Cyclone-Server will send the results and the complete logs of CI & CD workflow to users by email when the progress has finished
```
Comment (superxi911) [inline docs/developer-guide.md]: has been finished

File: docs/developer-guide.md [hunk 5]
```diff
@@ -1,5 +1,35 @@
-## Workflow
+# Developer Guide
+
+This section of the documentation contains a guide for users who want to contribute code or documentation to Cyclone.
+
+## The process of developing the Cyclone project
+
+Now the Cyclone project is hosted on Github, so we follow the [github flow](https://guides.github.com/introduction/flow/), you could learn it in 5 minutes:)
```
Comment (gaocegege) [inline docs/developer-guide.md]: I think "in" is more suitable?

**Comment by Wumengna0402** [comment]
因为我看了中文，表达的是花费5分钟的意思是嘛，in 5mins，会给人感觉是5分钟后。发自网易邮箱大师
            On 11/09/2016 19:00, Ce Gao wrote:@gaocegege commented on this pull request.



In docs/developer-guide.md:
> @@ -1,5 +1,35 @@
-## Workflow
+# Developer Guide
+
+This section of the documentation contains a guide for users who want to contribute code or documentation to Cyclone.
+
+## The process of developing the Cyclone project
+
+Now the Cyclone project is hosted on Github, so we follow the [github flow](https://guides.github.com/introduction/flow/), you could learn it in 5 minutes:)

I think "in" is more suitable?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/caicloud/cyclone","title":"caicloud/cyclone","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/caicloud/cyclone"}},"updates":{"snippets":[{"icon":"PERSON","message":"@gaocegege commented on #37"}],"action":{"name":"View Pull Request","url":"https://github.com/caicloud/cyclone/pull/37"}}}

File: docs/developer-guide.md (hunk 3 above)
Comment (gaocegege) [inline docs/developer-guide.md]: 端口转发 should be "port forwarding", I think = =

**Comment by Wumengna0402** [comment]
如果你要表达的是5分钟以内，那么考虑使用within？发自网易邮箱大师
            On 11/09/2016 19:00, Ce Gao wrote:@gaocegege commented on this pull request.



In docs/developer-guide.md:
> @@ -1,5 +1,35 @@
-## Workflow
+# Developer Guide
+
+This section of the documentation contains a guide for users who want to contribute code or documentation to Cyclone.
+
+## The process of developing the Cyclone project
+
+Now the Cyclone project is hosted on Github, so we follow the [github flow](https://guides.github.com/introduction/flow/), you could learn it in 5 minutes:)

I think "in" is more suitable?

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/caicloud/cyclone","title":"caicloud/cyclone","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/caicloud/cyclone"}},"updates":{"snippets":[{"icon":"PERSON","message":"@gaocegege commented on #37"}],"action":{"name":"View Pull Request","url":"https://github.com/caicloud/cyclone/pull/37"}}}

**Comment by Wumengna0402** [comment]
啊对，用port forwarding，my bad.发自网易邮箱大师
            On 11/09/2016 19:03, Ce Gao wrote:@gaocegege commented on this pull request.



In docs/developer-guide.md:
> +
+Now the Cyclone project is hosted on Github, so we follow the [github flow](https://guides.github.com/introduction/flow/), you could learn it in 5 minutes:)
+
+## Setting up your dev environment
+
+We have writen some bash scripts to help you start contributing code or documentation to the Cyclone project.
+
+If you want to start a cyclone server for dev, you could run:
+
+```shell
+./scripts/local-up.sh
+```
+
+The script would run all the dependencies in docker containers, compile and run cyclone server **locally**. This way is more hackable than run all services in containers.
+
+Notice that if your docker daemon is running in a docker machine, maybe you need to do some additional work, such as port forwarding and so on.

端口转发 should be "port forwarding", I think = =

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/caicloud/cyclone","title":"caicloud/cyclone","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/caicloud/cyclone"}},"updates":{"snippets":[{"icon":"PERSON","message":"@gaocegege commented on #37"}],"action":{"name":"View Pull Request","url":"https://github.com/caicloud/cyclone/pull/37"}}}

**Comment by gaocegege** [comment]
PTAL

**Comment by superxi911** [comment]
LGTM

**Comment by gaocegege** [comment]
@Wumengna0402 Thanks for your review :)

**Comment by Wumengna0402** [comment]
My pleasure～发自网易邮箱大师
            On 11/09/2016 19:24, Ce Gao wrote:@Wumengna0402 Thanks for your review :)

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/caicloud/cyclone","title":"caicloud/cyclone","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/caicloud/cyclone"}},"updates":{"snippets":[{"icon":"PERSON","message":"@gaocegege in #37: @Wumengna0402 Thanks for your review :)"}],"action":{"name":"View Pull Request","url":"https://github.com/caicloud/cyclone/pull/37#issuecomment-259391951"}}}

**Pull request merged**